src/socialMediaActivity.js

```diff
@@ -28,17 +29,155 @@ permissions.check({
  */
```

Comment:
The `permissions.check` call above should go in the add listener callback functions, with the right match pattern and permissions (`webRequestBlocking`?) for the event type and listener options.